src/server/scripts/Northrend/Naxxramas/boss_thaddius.cpp

```diff
@@ -384,144 +700,354 @@ class npc_feugen : public CreatureScript
-                    if ((*ihit)->GetGUID() != GetCaster()->GetGUID())
-                        if (Player* target = (*ihit)->ToPlayer())
-                            if (target->HasAura(GetTriggeringSpell()->Id))
-                                ++count;
+                if (!UpdateVictim())
+                    return;
 
-                if (count)
+                if (magneticPullTimer <= uiDiff)
                 {
-                    uint32 spellId = 0;
+                    DoCast(me, SPELL_MAGNETIC_PULL);
+                    magneticPullTimer = 30 * IN_MILLISECONDS;
+                }
+                else magneticPullTimer -= uiDiff;
+            
+                if (staticFieldTimer <= uiDiff)
+                {
+                    DoCast(me, SPELL_FEUGEN_STATICFIELD);
+                    staticFieldTimer = 5 * IN_MILLISECONDS;
+                }
+                else staticFieldTimer -= uiDiff;
 
-                    if (GetSpellInfo()->Id == SPELL_POSITIVE_CHARGE)
-                        spellId = SPELL_POSITIVE_CHARGE_STACK;
-                    else // if (GetSpellInfo()->Id == SPELL_NEGATIVE_CHARGE)
-                        spellId = SPELL_NEGATIVE_CHARGE_STACK;
+                DoMeleeAttackIfReady();
+            }
 
-                    GetCaster()->SetAuraStack(spellId, GetCaster(), count);
-                }
+        private:
+            Creature* myCoil()
+            {
+                Creature* coil = NULL;
+                if (_myCoil)
+                    coil = ObjectAccessor::GetCreature(*me, _myCoil);
+                if (!coil)
+                    if (coil = me->FindNearestCreature(NPC_TESLA, 1000.0f, true))
+                    {
+                        _myCoil = coil->GetGUID();
+                        coil->SetReactState(REACT_PASSIVE);
+                    }
+                return coil;
+            }
+
+            GameObject* myCoilGO()
+            {
+                GameObject* coil = NULL;
+                if (_myCoilGO)
+                    coil = ObjectAccessor::GetGameObject(*me, _myCoilGO);
+                if (!coil)
+                    if (coil = me->FindNearestGameObject(GO_CONS_NOX_TESLA_FEUGEN, 1000.0f))
+                        _myCoilGO = coil->GetGUID();
+                return coil;
+            }
+            InstanceScript* instance;
+
+            uint32 magneticPullTimer;
+            uint32 staticFieldTimer;
+
+            ObjectGuid _myCoil;
+            ObjectGuid _myCoilGO;
+       
+            bool isOverloading;
+    };
+};
+
+class npc_tesla : public CreatureScript
+{
+public:
+    npc_tesla() : CreatureScript("npc_tesla") { }
+
+    CreatureAI* GetAI(Creature* creature) const override
+    {
+        return GetInstanceAI<npc_teslaAI>(creature);
+    }
+
+    class npc_teslaAI : public ScriptedAI
+    {
+        public:
+            npc_teslaAI(Creature* creature) : ScriptedAI(creature) { }
```

Comment:
NullAI still has evade behavior, at least it kept cancelling its current spell cast on reset.